src/symbol.rs

```diff
@@ -4,13 +4,13 @@ use std::borrow::Borrow;
 use std::cmp::Ordering;
 use std::fmt::{Display, Formatter};
 use std::hash::{Hash, Hasher};
-use std::rc::Rc;
+use std::sync::Arc;
 
 /// Stores or points to the text of a given [Symbol].
 #[derive(Debug, Eq)]
 enum SymbolText {
     // This Symbol refers to a string in the symbol table
-    Shared(Rc<str>),
+    Shared(Arc<str>),
```

Comment:
🗺️ Multiple users have expressed a desire to be able to store an `Element` in an `Arc` or send an `Element` across threads, which is currently impossible because `Rc` does not implement `Send`.

Using `Arc` imposes a higher overhead cost when creating or dropping `Symbol`s, but makes `Element`s much more flexible and eliminates a source of surprise from the API.